server-timing/class-perflab-server-timing.php

```diff
@@ -258,6 +258,10 @@ private function format_metric_header_value( Perflab_Server_Timing_Metric $metri
 		if ( is_float( $value ) ) {
 			$value = round( $value, 2 );
 		}
-		return sprintf( 'wp-%1$s;dur=%2$s', $metric->get_slug(), $value );
+
+		// See https://github.com/WordPress/performance/issues/955.
+		$name = str_replace( '/', '-', $metric->get_slug() );
```

Comment:
Looking up the `Server-Timing` field definition, it is [comprised](https://www.w3.org/TR/server-timing/#the-server-timing-header-field) as follows:

```
  Server-Timing             = #server-timing-metric
  server-timing-metric      = metric-name *( OWS ";" OWS server-timing-param )
  metric-name               = token
  server-timing-param       = server-timing-param-name OWS "=" OWS server-timing-param-value
  server-timing-param-name  = token
  server-timing-param-value = token / quoted-string
```

So `$name` here should be `metric-name` which is a `token`. The spec then links off to RFC 7230 which [defines](https://httpwg.org/specs/rfc7230.html#rfc.section.3.2.6) `token` as:

```
  token          = 1*tchar

  tchar          = "!" / "#" / "$" / "%" / "&" / "'" / "*"
                 / "+" / "-" / "." / "^" / "_" / "`" / "|" / "~" 
                 / DIGIT / ALPHA
                 ; any VCHAR, except delimiters
```

So should more characters be replaced with `-` than just `/`? Maybe something like:

```suggestion
		$name = preg_replace( '/[^!#$%&\'*+\-.^_`|~0-9a-zA-Z]/', '-', $metric->get_slug() );
```